lib/grape/api.rb

```diff
@@ -126,16 +129,22 @@ def cascade?
     # will return an HTTP 405 response for any HTTP method that the resource
     # cannot handle.
     def add_head_not_allowed_methods_and_options_methods
-      methods_per_path = {}
+      routes_map = {}
 
       self.class.endpoints.each do |endpoint|
         routes = endpoint.routes
         routes.each do |route|
-          methods_per_path[route.route_path] ||= []
-          methods_per_path[route.route_path] << route.route_method
+          # using the :any shorthand produces [nil] for route methods, substitute all manually
+          route_settings = (routes_map[route.pattern.to_regexp] ||= {})
```

Comment:
I am not in love with this `||=` assignment inside the right hand side. It's just a little difficult to read.